### PR TITLE
feat(bytes): @endo/bytes package for platform-neutral byte operations

### PR DESCRIPTION
**Pull request by kriskowal**
## Description

Adds a new leaf package `@endo/bytes` providing platform-neutral byte operations on `Uint8Array`: `concatBytes`, `bytesEqual`, `bytesFromText`, `bytesToText`, `bytesFromImmutable`, `bytesToImmutable`, and `concatImmutables`. `TextEncoder` and `TextDecoder` are captured once at module load before any host-mutable state can interfere. The cross-conversion helpers between mutable `Uint8Array` and the immutable `ArrayBuffer` shape produced by the proposal-immutable-arraybuffer shim let passable bytes ride across vat boundaries.

The public surface uses per-export modules with kebab-case file names. Consumers import per-symbol from the corresponding subpath; there is no barrel module.

Migrates the `cli`, `daemon`, and `ocapn` packages to use `@endo/bytes` in place of their respective ad-hoc `Uint8Array` helpers and the previous `@endo/ocapn`-internal `buffer-utils` module. The `buffer-utils` module and its tests are retired; the coverage they provided is preserved by `@endo/bytes`'s own tests.

First release at `1.0.0` via a `major` changeset from the `0.1.0` workspace floor.

### Security Considerations

The package exposes no new authorities. `bytesFromText` and `bytesToText` use the host's `TextEncoder` and `TextDecoder`, captured once at module load before any host-mutable state can interfere. The encoder instances are not exposed to consumers. All exported functions are hardened.

### Scaling Considerations

`concatBytes` allocates a single output `Uint8Array` of the summed length and copies each input chunk in once. `bytesEqual` short-circuits on length mismatch and on first-differing byte; otherwise a single linear scan. None of the helpers introduce per-call allocations proportional to anything other than the inputs. The cross-mutability conversions (`bytesFromImmutable` and `bytesToImmutable`) copy once.

### Documentation Considerations

The package's `README.md` documents each export with a one-paragraph contract and a worked example. No separate docs-site changes are required; the package is a leaf utility with a small surface area.

### Testing Considerations

The package ships with tests across all three SES configurations (lockdown, unsafe, endo), covering empty input, alignment-sensitive lengths, multi-byte and non-BMP UTF-8, and round-trip composition through `concatBytes`. Tests for the retired `buffer-utils` module are migrated into the new package's suite so the round-trip and edge-case coverage carries forward.

### Compatibility Considerations

`@endo/bytes` is a new public API; no prior contract to preserve.

The consumer migrations in `cli`, `daemon`, and `ocapn` are internal refactors: each call site swaps a local ad-hoc helper for the equivalent `@endo/bytes` function. The external behavior of those packages is unchanged. The removal of `@endo/ocapn/src/buffer-utils.js` is a non-breaking change because that path was never part of `@endo/ocapn`'s public exports.

### Upgrade Considerations

No live-system upgrade impact. `@endo/bytes` is a leaf package with no persisted state. The consumer-package refactors do not change any wire format, file format, or persisted data.

### CODE REVIEW

File: packages/bytes/src/to-immutable.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ *
+ * @param {Uint8Array} view
+ * @returns {ArrayBuffer} A hardened immutable `ArrayBuffer`.
+ */
+export const bytesToImmutable = view => {
+  const immutable = sliceBufferToImmutable(
+    /** @type {ArrayBuffer} */ (view.buffer),
+    view.byteOffset,
+    view.byteOffset + view.byteLength,
+  );
```
[the remark this answers is not in the document]
Comment (kriskowal) [reply packages/bytes/src/to-immutable.js:27]: For this, I am content discouraging this misuse with the type system.

**Review by kumavis — APPROVED** [review]
thanks. ocapn looks good
